mistralrs-core/src/pipeline/mod.rs

```diff
@@ -458,10 +397,11 @@ pub struct GeneralMetadata {
     pub repeat_last_n: usize,
     pub tok_trie: Arc<TokTrie>,
     pub has_no_kv_cache: bool,
-    pub is_xlora: bool,
     pub num_hidden_layers: usize,
     pub eos_tok: Vec<u32>,
-    pub is_lora: bool,
+    pub kind: ModelKind,
+    // TODO: Replace is_xlora queries to check via kind instead:
```

Comment:
Do you want to do this in this PR or a future one?

Reply:
I can do it in a follow-up PR if you want to get this merged. Either way I can tackle it after I've had some rest, it's my cut off point for the day now :)

Reply:
Sounds good, I'll merge this.